app/uk/gov/hmrc/plasticpackagingtax/returns/controllers/actions/AuthAction.scala

```diff
@@ -0,0 +1,105 @@
+
+  private val authData =
+    credentials and name and email and externalId and internalId and affinityGroup and allEnrolments and
+      agentCode and confidenceLevel and nino and saUtr and dateOfBirth and agentInformation and groupIdentifier and
+      credentialRole and mdtpInformation and itmpName and itmpDateOfBirth and itmpAddress and credentialStrength and loginTimes
+
+  override def invokeBlock[A](
+    request: Request[A],
+    block: AuthenticatedRequest[A] => Future[Result]
+  ): Future[Result] = {
+    implicit val hc: HeaderCarrier =
+      HeaderCarrierConverter.fromRequestAndSession(request, request.session)
+    authorised()
+      .retrieve(authData) {
+        case credentials ~ name ~ email ~ externalId ~ internalId ~ affinityGroup ~ allEnrolments ~ agentCode ~
+            confidenceLevel ~ authNino ~ saUtr ~ dateOfBirth ~ agentInformation ~ groupIdentifier ~
+            credentialRole ~ mdtpInformation ~ itmpName ~ itmpDateOfBirth ~ itmpAddress ~ credentialStrength ~ loginTimes =>
+          val identityData = IdentityData(internalId,
+                                          externalId,
+                                          agentCode,
+                                          credentials,
+                                          Some(confidenceLevel),
+                                          authNino,
+                                          saUtr,
+                                          name,
+                                          dateOfBirth,
+                                          email,
+                                          Some(agentInformation),
+                                          groupIdentifier,
+                                          credentialRole.map(res => res.toJson.toString()),
+                                          mdtpInformation,
+                                          itmpName,
+                                          itmpDateOfBirth,
+                                          itmpAddress,
+                                          affinityGroup,
+                                          credentialStrength,
+                                          Some(loginTimes)
+          )
+
+          val pptLoggedInUser = SignedInUser(allEnrolments, identityData)
+          block(
+            new AuthenticatedRequest(request, pptLoggedInUser, getPptEnrolmentId(allEnrolments))
+          )
+      }
+  }
+
+  private def getPptEnrolmentId(enrolments: Enrolments): Option[String] =
+    getPptEnrolment(enrolments) match {
+      case Some(enrolmentId) => Option(enrolmentId).filter(_.value.trim.nonEmpty).map(_.value)
+      case None              => Option.empty
+    }
+
+  private def getPptEnrolment(enrolments: Enrolments): Option[EnrolmentIdentifier] =
+    enrolments
+      .getEnrolment(AuthAction.pptEnrolmentKey)
+      .flatMap(_.getIdentifier(AuthAction.pptEnrolmentIdentifierName))
+
+}
+
+object AuthAction {
+  val pptEnrolmentKey            = "HMRC-PPT-ORG"
```

Comment:
WE need to change this on the registrations service as well... otherwise we won't be able to jump between MS'

Reply:
yeap, story is created, next to pick up in the sprint